sormas-app/app/src/androidTest/java/de/symeda/sormas/app/FacilityBackendTest.java

```diff
@@ -45,16 +52,45 @@ public void initTest() {
 	}
 
 	@Test
-	public void shouldGetHealthFacilitiesByDistrict() {
+	public void shouldGetHealthFacilitiesByDistrict() throws SQLException {
 		District district = DatabaseHelper.getDistrictDao().queryUuid(TestHelper.DISTRICT_UUID);
+		Region region = district.getRegion();
 
-		// There should be exactly one health facility in this district - labs are not included
+		// There should be exactly one health facility and one laboratory in this district
 		assertThat(
-			DatabaseHelper.getFacilityDao().getActiveHealthFacilitiesByDistrictAndType(district, FacilityType.HOSPITAL, false, false).size(),
+			DatabaseHelper.getFacilityDao().getActiveHealthFacilitiesByDistrictAndType(district, FacilityType.LABORATORY, false, false).size(),
 			is(1));
 
 		assertThat(
-			DatabaseHelper.getFacilityDao().getActiveHealthFacilitiesByDistrictAndType(district, FacilityType.HOSPITAL, true, true).size(),
-			is(3));
+			DatabaseHelper.getFacilityDao().getActiveHealthFacilitiesByDistrictAndType(district, FacilityType.HOSPITAL, false, false).size(),
+			is(1));
+
+		List<Facility> activeHealthFacilitiesByDistrictAndType =
+			DatabaseHelper.getFacilityDao().getActiveHealthFacilitiesByDistrictAndType(district, null, true, true);
+		assertThat(activeHealthFacilitiesByDistrictAndType.size(), is(2));
+
+		Facility otherFacility = new Facility();
```

Comment:
Please move the creation of OTHER/NONE facilities into a new method in TestHelper that is called by TestHelper.insertInfrastructureData. To be realistic, we should always have those available.

I'd like to have one additional assertion that tests (district, FacilityType.HOSPITAL, true, true) -> should be 3

Reply:
done.